mvc/views/razor.rst

```diff
@@ -1,8 +1,481 @@
+inside of an attribute.
+
+
+
+.. code-block:: C#
+
+  <input custom="@true foo"/>
+  <input custom="@null @null"/>
+
+.. code-block:: HTML
+
+  <input custom="True foo"/>
+  <input custom=""/>
+
+Layouts
+-------
+To get a consistent look on a page it is often desirable to reuse HTML. This can
+be done through layouts. A layout is a template that Razor will use to construct
+each page. A Layout contains ``@RenderBody()`` and markup.
+
+
+Layout
+~~~~~~
+.. code-block:: HTML
+
+  <html>
+    <head>
+      <title>title</title>
+    </head>
+    <body>
+      @RenderBody()
+    </body>
+  </html>
+
+To use the layout the views layout must be set to the specified layout. This is
+done by setting the ``Layout`` variable in the view.
+
+View
+~~~~
+.. code-block:: HTML
+
+  @{Layout = "~/Views/SomeLayout.cshtml";}
+  <h1>Some Title</h1>
+  <p>Some Content</p>
+
+When rendering the page Razor takes the layout and replaces ``@RenderBody()``
+with the contents of the view. By default a view will attempt to use the layout
+at ``~/Views/Shared/_Layout.cshmtl``.
+
+Output
+~~~~~~
+
+.. code-block:: HTML
+
+  <html>
+    <head>
+      <title>title</title>
+    </head>
+    <body>
+      <h1>Some Title</h1>
+      <p>Some Content</p>
+    </body>
+  </html>
+
+Razor Directives
+----------------
+A Razor directive is a bit of code that isn't valid HTML or C# code, instead
+it provides instruction to Razor on how to perform certain actions.
+
+
+@model
+~~~~~~
+The ``@model`` directive is a way to indicate what model the view is using. Just
+place it at the top of the view along with the type of the model.
+
+.. code-block:: C#
+
+  @model Example.Models.Book
+
+Later on in the document you can use implicit and explicit expressions to render
+properties and method of that model via ``@Model.SomeProperty`` or
+``@Model.SomeMethod()``.
+
+.. code-block:: C#
+
+  ISBN: @Model.ISBN
+  Author: @Model.AuthorName()
+
+
+@using
+~~~~~~
+The ``@using`` directive is the Razor equivalent of the C# ``using`` statement.
+
+.. code-block:: C#
+
+  @using Example.Models
+
+
+@functions
+~~~~~~~~~~
+A ``@function`` block is a specific type of code block, that unlike a normal
+code block can contain functions. They are commonly used to contain
+reusable display related code inside of the view.
+
+.. code-block:: C#
+
+  @functions {
+    string doFoo(int times){
+      string out = "";
+      for(int i = 0; i < times; i++)
+        out += "Foo ";
+
+      return out;
+    }
+  }
+
+  <p>@doFoo(1)</p>
+  <h1>@doFoo(3)</h1>
+
+.. code-block:: HTML
+
+  <p>Foo </p>
+  <h1>Foo Foo </h1>
+
+
+@injects
+~~~~~~~~
+Razor now supports dependency injection. View specific services can be injected
+independent of the controller.
+
+.. code-block:: C#
+
+  @inject IInjectedServiceInterface InjectedServiceName
+
+A more in depth guide can be found here :doc:`dependency-injection`
+
+
+@await
+~~~~~~
+Razor can render the results of an asynchronous method call through
```

Comment:
I renamed this from Async to @await, it seems more appropriate. In addition is this actually a razor directive?